test/Microsoft.AspNet.Razor.Runtime.Test/TagHelpers/TagHelperExecutionContextTest.cs

```diff
@@ -153,14 +154,14 @@ public void HtmlAttributes_IgnoresCase(string originalName, string updatedName)
         {
             // Arrange
             var executionContext = new TagHelperExecutionContext("p", selfClosing: false);
-            executionContext.HTMLAttributes[originalName] = "hello";
+            executionContext.HTMLAttributes[originalName].Value = "hello";
```

Comment:
hmm, `public static implicit operator TagHelperAttribute(string value)` may be worthwhile.  would work well without setters on `TagHelperAttribute` properties as well.